pkg/sql/drop_database.go

```diff
@@ -128,6 +129,12 @@ func (n *dropDatabaseNode) startExec(params runParams) error {
 	ctx := params.ctx
 	p := params.p
 
+	// Exit early with an error if the schema is undergoing a declarative schema
+	// change.
+	if n.dbDesc.GetDeclarativeSchemaChangerState() != nil {
```

Comment:
Why not `if n.dbDesc.HasConcurrentSchemaChanges() {` instead? Same comment applies for other legacy DROPs.